tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/DynamoDb/DeleteItemAsyncIntegration.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="DeleteItemAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.DynamoDb
+{
+    /// <summary>
+    /// AWSSDK.DynamoDBv2 DeleteItemAsync CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.DynamoDBv2",
+        TypeName = "Amazon.DynamoDBv2.AmazonDynamoDBClient",
+        MethodName = "DeleteItemAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1[Amazon.DynamoDBv2.Model.DeleteItemResponse]",
+        ParameterTypeNames = new[] { "Amazon.DynamoDBv2.Model.DeleteItemRequest", ClrNames.CancellationToken },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsDynamoDbCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class DeleteItemAsyncIntegration
+    {
+        private const string Operation = "DeleteItem";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TDeleteItemRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the DynamoDB operation</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TDeleteItemRequest>(TTarget instance, TDeleteItemRequest request, CancellationToken cancellationToken)
+            where TDeleteItemRequest : IAmazonDynamoDbRequestWithTableName, IDuckType
+        {
+            if (request.Instance is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var scope = AwsDynamoDbCommon.CreateScope(Tracer.Instance, Operation, out AwsDynamoDbTags tags);
+            tags.TableName = request.TableName;
+            var span = scope.Span;
+            span.ResourceName = $"{span.ResourceName} {request.TableName}";
```

Comment:
nitpick: maybe refactor this into a helper method? and we can also use it in the batch case?
a bit blocking: check null on tags here? or maybe we add a big try catch just in case any error ? (I forgot if errors here can break the customer code... just by looking at it, it seems possible)

Reply:
Will refactor this into a helper method to tag when we now `TableName` is present. And yeah, probably we can use it in batch cases too.

Around tags being null, I don't think this will ever happen, since we always create tags, even is scope fails. And errors will not break customer code, since all of this is wrapped, and it will continue as normal.

Reply:
I think Joey's correct, currently there's a case when tags could be null, which is when the AWS DynamoDb integration has been disabled. I think a refactor is a good idea and could help make sure we don't get a null reference exception in that scenario.

We also want to be as careful as possible as throwing an exception would result in this integration being disabled by our automatic instrumentation machinery, as a safety precaution

Reply:
Sounds good, will take it in consideration for the tagging refactoring.